src/main/java/it/unitn/ds1/Replica.java

```diff
@@ -247,7 +308,46 @@ private void onReadMsg(ReadMsg msg) {
     }
 
     private void onHeartbeatMsg(HeartbeatMsg msg) {
+        if (this.isCoordinator) {
+            // Sends an heartbeat to all other replicas
+            this.multicast(msg);
+        } else {
+            // Reset lastContact
```

Comment:
These comments are useless, you are just putting the name of the method itself as a comment.
Instead, explain what the method does before, like:
/**
* For the coordinator: it sends a response to all replicas when receiving a request, to signal that it's still alive
* For the replicas: they received the signal from the coordinator, so we know that it's still alive
*/

Reply:
Yeah I can review those comments